addons/mail/controllers/mailbox.py

```diff
@@ -21,4 +21,12 @@ def discuss_history_messages(self, before=None, after=None, limit=30):
     @http.route("/mail/starred/messages", methods=["POST"], type="json", auth="user")
     def discuss_starred_messages(self, before=None, after=None, limit=30):
         domain = [("starred_partner_ids", "in", [request.env.user.partner_id.id])]
-        return request.env["mail.message"]._message_fetch(domain, before, after, limit=limit).message_format()
+        starred_messages = request.env["mail.message"]
+        for message in request.env["mail.message"]._message_fetch(domain, before, after, limit=limit):
+            if message.model != "discuss.channel":
+                starred_messages |= message
+                continue
+            channel = request.env["discuss.channel"].browse(message.res_id)
+            if request.env.user.partner_id in channel.channel_partner_ids:
+                starred_messages |= message
+        return starred_messages.message_format()
```

Comment:
We should still not have changes about the way we compute starred messages in the PR handling the forward-port of the channel leave.

If these changes have to be done for starred, it should be done on its own PR, targeting the first version with the issue.